ui/pages/asset/components/asset-price.tsx

```diff
@@ -0,0 +1,149 @@
+import React, { useState, forwardRef, useImperativeHandle } from 'react';
+import {
+  BackgroundColor,
+  BorderRadius,
+  Display,
+  TextColor,
+  TextVariant,
+} from '../../../helpers/constants/design-system';
+import { Box, Text } from '../../../components/component-library';
+import { formatCurrency } from '../../../helpers/utils/confirm-tx.util';
+import { getPricePrecision, shortDateFormatter } from '../util';
+
+const chartUp = (
+  <svg
+    className="chart-up"
+    width="12"
+    height="12"
+    viewBox="0 0 12 12"
+    fill="none"
+    xmlns="http://www.w3.org/2000/svg"
+  >
+    <path
+      d="M9.75 3.8125L6.25 7.4875L4.91667 5.3875L2.25 8.1875"
+      stroke-linecap="round"
+      stroke-linejoin="round"
+    />
+    <path
+      d="M8.08398 3.8125H9.75065V5.5625"
+      stroke-linecap="round"
+      stroke-linejoin="round"
+    />
+  </svg>
+);
+const chartDown = (
+  <svg
+    className="chart-down"
+    width="12"
+    height="12"
+    viewBox="0 0 12 12"
+    fill="none"
+    xmlns="http://www.w3.org/2000/svg"
+  >
+    <path
+      d="M9.75 8.1875L6.25 4.5125L4.91667 6.6125L2.25 3.8125"
+      stroke-linecap="round"
+      stroke-linejoin="round"
+    />
+    <path
+      d="M8.08398 8.1875H9.75065V6.4375"
+      stroke-linecap="round"
+      stroke-linejoin="round"
+    />
+  </svg>
+);
+
+// A component that shows the price of an asset at a
+// certain time, along with the delta from a previous price.
+const AssetPrice = forwardRef(
+  (
+    props: {
+      loading: boolean;
+      currency: string;
+      price?: number;
+      date: number;
+      comparePrice?: number;
+    },
+    ref,
+  ) => {
+    const [{ price, date }, setPrice] = useState({
+      price: props.price,
+      date: props.date,
+    });
+    useImperativeHandle(ref, () => ({ setPrice }));
+
+    const { loading, currency, comparePrice } = props;
+    const priceDelta =
+      price !== undefined && comparePrice !== undefined
+        ? price - comparePrice
+        : undefined;
+
+    return (
+      <Box>
+        <Text
+          style={{ width: '100px' }}
```

Comment:
This style is `100px` rather than `100%`.  It sets the size of the loading skeleton.  It's that magic number because it roughly matches the width of a 4 digit number.


<img width="300" alt="Screenshot 2024-03-14 at 10 19 06 AM" src="https://github.com/MetaMask/metamask-extension/assets/3500406/3d5e8bac-c8fe-4ce4-bd7f-fc3c877e9501">